### PR TITLE
Support for serving files in http2

### PR DESCRIPTION
**Pull request by ypiguet-epfl**

### CODE REVIEW

File: switches/http2/main.cpp [hunk 1]
```diff
@@ -4,16 +4,16 @@
 		Stephane Magnenat <stephane at magnenat dot net>
 		(http://stephane.magnenat.net)
 		and other contributors, see authors.txt for details
-
```
Comment (stephanemagnenat) [inline switches/http2/main.cpp]: Whitespace changes, please remove

File: switches/http2/main.cpp [hunk 2]
```diff
@@ -4,16 +4,16 @@
 		Stephane Magnenat <stephane at magnenat dot net>
 		(http://stephane.magnenat.net)
 		and other contributors, see authors.txt for details
-	
+
 	This program is free software: you can redistribute it and/or modify
 	it under the terms of the GNU Lesser General Public License as published
 	by the Free Software Foundation, version 3 of the License.
-
```
Comment (stephanemagnenat) [inline switches/http2/main.cpp]: Whitespace changes, please remove

File: switches/http2/main.cpp [hunk 3]
```diff
@@ -4,16 +4,16 @@
...
 	This program is free software: you can redistribute it and/or modify
 	it under the terms of the GNU Lesser General Public License as published
 	by the Free Software Foundation, version 3 of the License.
-	
+
 	This program is distributed in the hope that it will be useful,
 	but WITHOUT ANY WARRANTY; without even the implied warranty of
 	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 	GNU Lesser General Public License for more details.
-
```
Comment (stephanemagnenat) [inline switches/http2/main.cpp]: Whitespace changes, please remove

File: switches/http2/HttpInterfaceHandlers.cpp [hunk 4]
```diff
@@ -475,3 +477,86 @@ void VariableOrEventHandler::parseJsonForm(std::string content, std::vector<std:
...
+	{ "xml", "application/xml" }
+};
+
+void FileHandler::handleRequest(HttpRequest *request, const std::vector<std::string>& tokens)
+{
+	std::string path = filePath(request);
+	std::ifstream f(path.c_str());
+	if (f.good())
+	{
+		std::string type = suffixMapping[getSuffix(path)];
```
Comment (stephanemagnenat) [inline switches/http2/HttpInterfaceHandlers.cpp]: If suffix from path is not in suffixMapping it is created, with an empty string as value; is it really what you want?

File: switches/http2/HttpInterfaceHandlers.cpp [hunk 5]
```diff
@@ -475,3 +477,86 @@ void VariableOrEventHandler::parseJsonForm(std::string content, std::vector<std:
...
+		if (!type.empty())
+		{
+			std::stringbuf content;
+			while (!f.eof())
+			{
+				#define bufferSize 1024
```
Comment (stephanemagnenat) [inline switches/http2/HttpInterfaceHandlers.cpp]: Use `const size_t` instead of local define.

**Comment by stephanemagnenat** [comment]
Looks good to merge (lgtm). Please rebase into a single commit and I'll merge.

**Comment by mbonani** [comment]
All checks fails:
Cloning the remote Git repository
Cloning repository https://github.com/aseba-community/dashel.git
 > git init /Volumes/jenkins/workspace/GitHub-Aseba_PR-647-E3UYWBG2BEYGIUT2ULOKTZ6QSRCRAXMOLIA7OPM2WUH7JS4DXC6A/externals/dashel # timeout=10
Fetching upstream changes from https://github.com/aseba-community/dashel.git
 > git --version # timeout=10
 > git fetch --tags --progress https://github.com/aseba-community/dashel.git +refs/heads/*:refs/remotes/origin/*
 > git config remote.origin.url https://github.com/aseba-community/dashel.git # timeout=10
 > git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git config remote.origin.url https://github.com/aseba-community/dashel.git # timeout=10
Fetching upstream changes from https://github.com/aseba-community/dashel.git
 > git fetch --tags --progress https://github.com/aseba-community/dashel.git +refs/heads/*:refs/remotes/origin/*
 > git rev-parse refs/remotes/origin/null^{commit} # timeout=10
 > git rev-parse refs/remotes/origin/origin/null^{commit} # timeout=10
 > git rev-parse origin/null^{commit} # timeout=10
Couldn't find any revision to build. Verify the repository and branch configuration for this job.

Is there a problem in the configuration or this is still not working because #581 was merged after? Is possible to relaunch manually the test-build?

**Comment by mbonani** [comment]
All checks fails:
Cloning the remote Git repository
Cloning repository https://github.com/aseba-community/dashel.git
 > git init /Volumes/jenkins/workspace/GitHub-Aseba_PR-647-E3UYWBG2BEYGIUT2ULOKTZ6QSRCRAXMOLIA7OPM2WUH7JS4DXC6A/externals/dashel # timeout=10
Fetching upstream changes from https://github.com/aseba-community/dashel.git
 > git --version # timeout=10
 > git fetch --tags --progress https://github.com/aseba-community/dashel.git +refs/heads/*:refs/remotes/origin/*
 > git config remote.origin.url https://github.com/aseba-community/dashel.git # timeout=10
 > git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git config remote.origin.url https://github.com/aseba-community/dashel.git # timeout=10
Fetching upstream changes from https://github.com/aseba-community/dashel.git
 > git fetch --tags --progress https://github.com/aseba-community/dashel.git +refs/heads/*:refs/remotes/origin/*
 > git rev-parse refs/remotes/origin/null^{commit} # timeout=10
 > git rev-parse refs/remotes/origin/origin/null^{commit} # timeout=10
 > git rev-parse origin/null^{commit} # timeout=10
Couldn't find any revision to build. Verify the repository and branch configuration for this job.

Is there a problem in the configuration or this is still not working because #581 was merged after? Is possible to relaunch manually the test-build?

**Pull request merged**